--- trunk.orig/boost/date_time/date_formatting.hpp
+++ trunk/boost/date_time/date_formatting.hpp
@@ -79,7 +79,13 @@
     {
       typedef typename ymd_type::month_type month_type;
       std::basic_ostringstream<charT> ss;
+
+      // Temporarily switch to classic locale to prevent possible formatting
+      // of year with comma or other character (for example 2,008).
+      ss.imbue(std::locale::classic());
       ss << ymd.year;
+      ss.imbue(std::locale::locale());
+
       if (format_type::has_date_sep_chars()) {
         ss << format_type::month_sep_char();
       }
--- trunk.orig/boost/date_time/posix_time/time_formatters.hpp
+++ trunk/boost/date_time/posix_time/time_formatters.hpp
@@ -144,6 +144,9 @@
         date_time::absolute_value(td.fractional_seconds());
 #endif
       if (frac_sec != 0) {
+        // Temporarily switch to classic locale to prevent possible formatting
+        // of frac_sec with comma or other character (for example 123,456).
+        ss.imbue(std::locale::classic());
         ss  << "." << std::setw(time_duration::num_fractional_digits())
             << std::setfill(fill_char)
           
@@ -153,6 +156,7 @@
 #else
         << frac_sec;
 #endif
+        ss.imbue(std::locale::locale());
       }
     }// else
     return ss.str();
