Index: boost_1_72_0/libs/mpi/build/__init__.py
===================================================================
--- boost_1_72_0.orig/libs/mpi/build/__init__.py
+++ boost_1_72_0/libs/mpi/build/__init__.py
@@ -5,6 +5,8 @@ if sys.platform == 'linux2':
     sys.setdlopenflags(dl.RTLD_NOW|dl.RTLD_GLOBAL)
     import mpi
     sys.setdlopenflags(flags)
+if sys.platform == 'linux':
+    from . import mpi
 else:
     import mpi
 
