Fix for http://svn.boost.org/trac/boost/ticket/1617

--- boost1.36-1.36.0.orig/libs/spirit/classic/test/mix_and_match_trees.cpp
+++ boost1.36-1.36.0/libs/spirit/classic/test/mix_and_match_trees.cpp
@@ -54,7 +54,7 @@
 
         typedef rule<ast_scanner> ast_rule;
         typedef rule<pt_scanner> pt_rule;
-        typedef rule<Scanner> rule;
+        typedef rule<Scanner> rule_;
 
         definition(my_grammar const & /* self */)
         {
@@ -62,12 +62,12 @@
             start_ = gen_pt_node_d[ pt_rule_ ];
         }
 
-        rule const & start() const
+        rule_ const & start() const
         {
             return start_;
         }
 
-        rule start_;
+        rule_ start_;
         ast_rule ast_rule_;
         pt_rule pt_rule_;
     };
--- boost1.36-1.36.0.orig/libs/spirit/classic/test/numerics_tests.cpp
+++ boost1.36-1.36.0/libs/spirit/classic/test/numerics_tests.cpp
@@ -11,6 +11,7 @@
 #include <boost/spirit/include/classic_assign_actor.hpp>
 #include <iostream>
 #include <boost/detail/lightweight_test.hpp>
+#include <climits>
 
 using namespace std;
 using namespace BOOST_SPIRIT_CLASSIC_NS;
