--- boost1.65.1-1.65.1+dfsg.orig/tools/build/src/tools/pathscale.jam
+++ boost1.65.1-1.65.1+dfsg/tools/build/src/tools/pathscale.jam
@@ -89,8 +89,6 @@
 flags pathscale.compile OPTIONS <debug-symbols>on : -ggdb ;
 flags pathscale.compile OPTIONS <profiling>on     : -pg ;
 flags pathscale.compile OPTIONS <link>shared      : -fPIC ;
-flags pathscale.compile OPTIONS <address-model>32 : -m32 ;
-flags pathscale.compile OPTIONS <address-model>64 : -m64 ;
 
 flags pathscale.compile USER_OPTIONS <cflags> ;
 flags pathscale.compile.c++ USER_OPTIONS <cxxflags> ;
