diff -ur boost-1.33.0/debian/rules boost-1.33.0.st_mt/debian/rules
--- boost-1.33.0/debian/rules	2005-10-24 21:06:17.000000000 +0200
+++ boost-1.33.0.st_mt/debian/rules	2005-10-24 21:03:38.000000000 +0200
@@ -17,6 +17,8 @@
 # Boost libraries encode build information in the resulting library
 # filename: toolset, threading, runtime, and version.
 #
+release_suffix_st = -gcc-$(boost_version)
+debug_suffix_st = -gcc-d-$(boost_version)
 release_suffix = -gcc-mt-$(boost_version)
 debug_suffix = -gcc-mt-d-$(boost_version)
 
@@ -33,7 +35,7 @@
 	-sTOOLS=gcc \
 	-sGCC=gcc \
 	-sGXX=g++ \
-	-sBUILD="debug release <define>_REENTRANT <threading>multi" \
+	-sBUILD="debug release <define>_REENTRANT <threading>single/multi" \
 	-sPYTHON_VERSION=2.3 \
 	-sPYTHON_ROOT=/usr
 
@@ -135,6 +137,10 @@
 	   debian/tmp/usr/lib/libboost_date_time$(release_suffix).so.$(SOVERSION)
 	dh_install -plibboost-dbg --autodest \
 	   debian/tmp/usr/lib/libboost_date_time$(debug_suffix).so.$(SOVERSION)
+	dh_install -plibboost-date-time$(SOVERSION) --autodest \
+	   debian/tmp/usr/lib/libboost_date_time$(release_suffix_st).so.$(SOVERSION)
+	dh_install -plibboost-dbg --autodest \
+	   debian/tmp/usr/lib/libboost_date_time$(debug_suffix_st).so.$(SOVERSION)
 	dh_link -plibboost-dbg \
 	   usr/lib/libboost_date_time$(debug_suffix).so.$(SOVERSION) \
 	   usr/lib/debug/usr/lib/libboost_date_time$(release_suffix).so.$(SOVERSION)
@@ -145,7 +151,9 @@
 	   debian/libboost-date-time-dev/usr/include/boost
 	dh_install -plibboost-date-time-dev --autodest \
 	   debian/tmp/usr/lib/libboost_date_time$(release_suffix).a \
-	   debian/tmp/usr/lib/libboost_date_time$(release_suffix).so
+	   debian/tmp/usr/lib/libboost_date_time$(release_suffix).so \
+	   debian/tmp/usr/lib/libboost_date_time$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_date_time$(release_suffix_st).so
 	dh_link -plibboost-date-time-dev \
 	   usr/lib/libboost_date_time$(release_suffix).a \
 	   usr/lib/libboost_date_time.a \
@@ -158,6 +166,10 @@
 	   debian/tmp/usr/lib/libboost_filesystem$(release_suffix).so.$(SOVERSION)
 	dh_install -plibboost-dbg --autodest \
 	   debian/tmp/usr/lib/libboost_filesystem$(debug_suffix).so.$(SOVERSION)
+	dh_install -plibboost-filesystem$(SOVERSION) --autodest \
+	   debian/tmp/usr/lib/libboost_filesystem$(release_suffix_st).so.$(SOVERSION)
+	dh_install -plibboost-dbg --autodest \
+	   debian/tmp/usr/lib/libboost_filesystem$(debug_suffix_st).so.$(SOVERSION)
 	dh_link -plibboost-dbg \
 	   usr/lib/libboost_filesystem$(debug_suffix).so.$(SOVERSION) \
 	   usr/lib/debug/usr/lib/libboost_filesystem$(release_suffix).so.$(SOVERSION)
@@ -167,6 +179,8 @@
 	mv debian/libboost-dev/usr/include/boost/filesystem \
 	   debian/libboost-filesystem-dev/usr/include/boost
 	dh_install -plibboost-filesystem-dev --autodest \
+	   debian/tmp/usr/lib/libboost_filesystem$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_filesystem$(release_suffix_st).so \
 	   debian/tmp/usr/lib/libboost_filesystem$(release_suffix).a \
 	   debian/tmp/usr/lib/libboost_filesystem$(release_suffix).so
 	dh_link -plibboost-filesystem-dev \
@@ -199,6 +213,10 @@
 	   debian/tmp/usr/lib/libboost_program_options$(release_suffix).so.$(SOVERSION)
 	dh_install -plibboost-dbg --autodest \
 	   debian/tmp/usr/lib/libboost_program_options$(debug_suffix).so.$(SOVERSION)
+	dh_install -plibboost-program-options$(SOVERSION) --autodest \
+	   debian/tmp/usr/lib/libboost_program_options$(release_suffix_st).so.$(SOVERSION)
+	dh_install -plibboost-dbg --autodest \
+	   debian/tmp/usr/lib/libboost_program_options$(debug_suffix_st).so.$(SOVERSION)
 	dh_link -plibboost-dbg \
 	   usr/lib/libboost_program_options$(debug_suffix).so.$(SOVERSION) \
 	   usr/lib/debug/usr/lib/libboost_program_options$(release_suffix).so.$(SOVERSION)
@@ -208,6 +226,8 @@
 	mv debian/libboost-dev/usr/include/boost/program_options* \
 	   debian/libboost-program-options-dev/usr/include/boost
 	dh_install -plibboost-program-options-dev --autodest \
+	   debian/tmp/usr/lib/libboost_program_options$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_program_options$(release_suffix_st).so \
 	   debian/tmp/usr/lib/libboost_program_options$(release_suffix).a \
 	   debian/tmp/usr/lib/libboost_program_options$(release_suffix).so
 	dh_link -plibboost-program-options-dev \
@@ -219,12 +239,16 @@
 
 	# package libboost-python$(SOVERSION)
 	dh_install -plibboost-python$(SOVERSION) --autodest \
+	   debian/tmp/usr/lib/libboost_python$(release_suffix_st).so.$(SOVERSION)
+	dh_install -plibboost-python$(SOVERSION) --autodest \
 	   debian/tmp/usr/lib/libboost_python$(release_suffix).so.$(SOVERSION)
 	mkdir -p debian/libboost-python$(SOVERSION)/usr/share/lintian/overrides
 	install -m644 \
 	   debian/libboost-python$(SOVERSION).lintian-overrides \
 	   debian/libboost-python$(SOVERSION)/usr/share/lintian/overrides/libboost-python$(SOVERSION)
 	dh_install -plibboost-dbg --autodest \
+	   debian/tmp/usr/lib/libboost_python$(debug_suffix_st).so.$(SOVERSION)
+	dh_install -plibboost-dbg --autodest \
 	   debian/tmp/usr/lib/libboost_python$(debug_suffix).so.$(SOVERSION)
 	dh_link -plibboost-dbg \
 	   usr/lib/libboost_python$(debug_suffix).so.$(SOVERSION) \
@@ -235,6 +259,8 @@
 	mv debian/libboost-dev/usr/include/boost/python* \
 	   debian/libboost-python-dev/usr/include/boost
 	dh_install -plibboost-python-dev --autodest \
+	   debian/tmp/usr/lib/libboost_python$(release_suffix_st).so
+	dh_install -plibboost-python-dev --autodest \
 	   debian/tmp/usr/lib/libboost_python$(release_suffix).so
 	dh_link -plibboost-python-dev \
 	   usr/lib/libboost_python$(release_suffix).so \
@@ -250,6 +276,10 @@
 	   debian/tmp/usr/lib/libboost_regex$(release_suffix).so.$(SOVERSION)
 	dh_install -plibboost-dbg --autodest \
 	   debian/tmp/usr/lib/libboost_regex$(debug_suffix).so.$(SOVERSION)
+	dh_install -plibboost-regex$(SOVERSION) --autodest \
+	   debian/tmp/usr/lib/libboost_regex$(release_suffix_st).so.$(SOVERSION)
+	dh_install -plibboost-dbg --autodest \
+	   debian/tmp/usr/lib/libboost_regex$(debug_suffix_st).so.$(SOVERSION)
 	dh_link -plibboost-dbg \
 	   usr/lib/libboost_regex$(debug_suffix).so.$(SOVERSION) \
 	   usr/lib/debug/usr/lib/libboost_regex$(release_suffix).so.$(SOVERSION)
@@ -259,6 +289,8 @@
 	mv debian/libboost-dev/usr/include/boost/*regex* \
 	   debian/libboost-regex-dev/usr/include/boost
 	dh_install -plibboost-regex-dev --autodest \
+	   debian/tmp/usr/lib/libboost_regex$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_regex$(release_suffix_st).so \
 	   debian/tmp/usr/lib/libboost_regex$(release_suffix).a \
 	   debian/tmp/usr/lib/libboost_regex$(release_suffix).so
 	dh_link -plibboost-regex-dev \
@@ -272,6 +304,8 @@
 	mv debian/libboost-dev/usr/include/boost/*serialization* \
 	   debian/libboost-serialization-dev/usr/include/boost
 	dh_install -plibboost-serialization-dev --autodest \
+	   debian/tmp/usr/lib/libboost_serialization$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_wserialization$(release_suffix_st).a \
 	   debian/tmp/usr/lib/libboost_serialization$(release_suffix).a \
 	   debian/tmp/usr/lib/libboost_wserialization$(release_suffix).a
 	dh_link -plibboost-serialization-dev \
@@ -286,6 +320,10 @@
 	   debian/tmp/usr/lib/libboost_signals$(release_suffix).so.$(SOVERSION)
 	dh_install -plibboost-dbg --autodest \
 	   debian/tmp/usr/lib/libboost_signals$(debug_suffix).so.$(SOVERSION)
+	dh_install -plibboost-signals$(SOVERSION) --autodest \
+	   debian/tmp/usr/lib/libboost_signals$(release_suffix_st).so.$(SOVERSION)
+	dh_install -plibboost-dbg --autodest \
+	   debian/tmp/usr/lib/libboost_signals$(debug_suffix_st).so.$(SOVERSION)
 	dh_link -plibboost-dbg \
 	   usr/lib/libboost_signals$(debug_suffix).so.$(SOVERSION) \
 	   usr/lib/debug/usr/lib/libboost_signals$(release_suffix).so.$(SOVERSION)
@@ -295,6 +333,8 @@
 	mv debian/libboost-dev/usr/include/boost/*signal* \
 	   debian/libboost-signals-dev/usr/include/boost
 	dh_install -plibboost-signals-dev --autodest \
+	   debian/tmp/usr/lib/libboost_signals$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_signals$(release_suffix_st).so \
 	   debian/tmp/usr/lib/libboost_signals$(release_suffix).a \
 	   debian/tmp/usr/lib/libboost_signals$(release_suffix).so
 	dh_link -plibboost-signals-dev \
@@ -306,10 +346,16 @@
 
 	# package libboost-test$(SOVERSION)
 	dh_install -plibboost-test$(SOVERSION) --autodest \
+	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(release_suffix_st).so.$(SOVERSION) \
+	   debian/tmp/usr/lib/libboost_test_exec_monitor$(release_suffix_st).so.$(SOVERSION) \
+	   debian/tmp/usr/lib/libboost_unit_test_framework$(release_suffix_st).so.$(SOVERSION) \
 	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(release_suffix).so.$(SOVERSION) \
 	   debian/tmp/usr/lib/libboost_test_exec_monitor$(release_suffix).so.$(SOVERSION) \
 	   debian/tmp/usr/lib/libboost_unit_test_framework$(release_suffix).so.$(SOVERSION)
 	dh_install -plibboost-dbg --autodest \
+	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(debug_suffix_st).so.$(SOVERSION) \
+	   debian/tmp/usr/lib/libboost_test_exec_monitor$(debug_suffix_st).so.$(SOVERSION) \
+	   debian/tmp/usr/lib/libboost_unit_test_framework$(debug_suffix_st).so.$(SOVERSION) \
 	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(debug_suffix).so.$(SOVERSION) \
 	   debian/tmp/usr/lib/libboost_test_exec_monitor$(debug_suffix).so.$(SOVERSION) \
 	   debian/tmp/usr/lib/libboost_unit_test_framework$(debug_suffix).so.$(SOVERSION)
@@ -326,6 +372,12 @@
 	mv debian/libboost-dev/usr/include/boost/test \
 	   debian/libboost-test-dev/usr/include/boost
 	dh_install -plibboost-test-dev --autodest \
+	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(release_suffix_st).so \
+	   debian/tmp/usr/lib/libboost_test_exec_monitor$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_test_exec_monitor$(release_suffix_st).so \
+	   debian/tmp/usr/lib/libboost_unit_test_framework$(release_suffix_st).a \
+	   debian/tmp/usr/lib/libboost_unit_test_framework$(release_suffix_st).so \
 	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(release_suffix).a \
 	   debian/tmp/usr/lib/libboost_prg_exec_monitor$(release_suffix).so \
 	   debian/tmp/usr/lib/libboost_test_exec_monitor$(release_suffix).a \
